test/extended/networking/vlan.go

```diff
@@ -0,0 +1,134 @@
+
+			g.By("creating bridge network attachment definition")
+			err := createNetworkAttachmentDefinition(
+				oc.AdminConfig(),
+				namespace,
+				bridge,
+				fmt.Sprintf(`{"cniVersion":"0.4.0","name":"%s","plugins":[{ "type": "bridge", "bridge": "%s"}]}`, bridge, bridge),
+			)
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("unable to create bridge network-attachment-definition"))
+
+			g.By("creating a vlan network attachment definition")
+			err = createNetworkAttachmentDefinition(
+				oc.AdminConfig(),
+				namespace,
+				vlan1,
+				fmt.Sprintf(vlanNadConfig, vlan1, pluginType, bridge, vlanId, ip1),
+			)
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("unable to create %s network-attachment-definition", pluginType))
+
+			err = createNetworkAttachmentDefinition(
+				oc.AdminConfig(),
+				namespace,
+				vlan2,
+				fmt.Sprintf(vlanNadConfig, vlan2, pluginType, bridge, vlanId, ip2),
+			)
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("unable to create %s network-attachment-definition", pluginType))
+
+			err = createNetworkAttachmentDefinition(
+				oc.AdminConfig(),
+				namespace,
+				otherVlan,
+				fmt.Sprintf(vlanNadConfig, otherVlan, pluginType, bridge, otherVlanId, otherIp),
+			)
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("unable to create %s network-attachment-definition", pluginType))
+
+			g.By("creating first pod and checking results")
+			exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName1, func(pod *kapiv1.Pod) {
+				vlanAnnotation := fmt.Sprintf("%s/%s@%s", namespace, vlan1, vlan1)
+				pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s", bridgeAnnotation, vlanAnnotation)}
+			})
+			pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Get(context.TODO(), podName1, metav1.GetOptions{})
+			o.Expect(err).ToNot(o.HaveOccurred())
+
+			networkStatusString, ok := pod.Annotations["k8s.v1.cni.cncf.io/network-status"]
+			o.Expect(ok).To(o.BeTrue())
+			o.Expect(networkStatusString).ToNot(o.BeNil())
+			networkStatuses := []nadtypes.NetworkStatus{}
+			o.Expect(json.Unmarshal([]byte(networkStatusString), &networkStatuses)).ToNot(o.HaveOccurred())
+			o.Expect(networkStatuses).To(o.HaveLen(3))
+			o.Expect(networkStatuses[2].Interface).To(o.Equal(vlan1))
+			o.Expect(networkStatuses[2].Name).To(o.Equal(fmt.Sprintf("%s/%s", namespace, vlan1)))
+
+			g.By("having a second pod pinging the first pod")
+			exutil.CreateExecPodOrFail(f.ClientSet, namespace, podName2, func(pod *kapiv1.Pod) {
+				vlanAnnotation := fmt.Sprintf("%s/%s@%s", namespace, vlan2, vlan2)
+				pod.ObjectMeta.Annotations = map[string]string{"k8s.v1.cni.cncf.io/networks": fmt.Sprintf("%s, %s", bridgeAnnotation, vlanAnnotation)}
+				pod.Spec.Containers[0].Command = []string{"/bin/bash", "-c", fmt.Sprintf("ping -c 1 %s", ip1)}
+			})
+
+			g.By("having another pod on different vlan unanble to ping the first pod")
```

Comment:
```suggestion
			g.By("having another pod on different vlan unable to ping the first pod")
```

Reply:
@mmirecki looks like this nit got missed.

Reply:
Thanks @bparees, I changed it. (@mmirecki is leaving RH and I'm taking over this PR).